File: compatibility-tests/src/test/java/nnl/rocks/kactoos/io/ReaderAsBytesTest.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package nnl.rocks.kactoos.io;

import nnl.rocks.kactoos.text.TextOf;
import org.hamcrest.MatcherAssert;
import org.hamcrest.Matchers;
import org.junit.Test;

import java.io.IOException;
import java.io.StringReader;

/**
 * Test case for {@link ReaderAsBytes}.
 *
 * @author Kirill (g4s8.public@gmail.com)
 * @version $Id: 779161b2c8c3e4979cd36ff54892439f498b690a $
 * @since 0.12
 * @checkstyle JavadocMethodCheck (500 lines)
 */
public final class ReaderAsBytesTest {

    @Test
    public void readsString() throws IOException {
        final String source = "hello, друг!";
        MatcherAssert.assertThat(
            "Can't read string through a reader",
            new TextOf(
                new ReaderAsBytes(
                    new StringReader(source)
                )
            ).asString(),
            Matchers.equalTo(source)
        );
    }

}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/io/ReaderAsBytesTest.kt

package nnl.rocks.kactoos.io

import nnl.rocks.kactoos.text.TextOf
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.Test

import java.io.IOException
import java.io.StringReader

/**
 * Test case for [ReaderAsBytes].
 *
 * @author Kirill (g4s8.public@gmail.com)
 * @version $Id: 779161b2c8c3e4979cd36ff54892439f498b690a $
 * @since 0.12
 * @checkstyle JavadocMethodCheck (500 lines)
 */
class ReaderAsBytesTest {

    @Test
    @Throws(IOException::class)
    fun readsString() {
        val source = "hello, друг!"
        MatcherAssert.assertThat(
                "Can't read string through a reader",
                TextOf(
                        ReaderAsBytes(
                                StringReader(source)
                        )
                ).asString(),
                Matchers.equalTo(source)
        )
    }
}
